### PR TITLE
runtime: Don' call bindUnmountContainerRootfs for devicemapper device

### PR DESCRIPTION
**Pull request by keloyang**
backport for https://github.com/kata-containers/runtime/pull/2915
buildContainerRootfs don't call bindMountContainerRootfs if container's rootfs use devicemapper
device in https://github.com/kata-containers/runtime/blob/master/virtcontainers/kata_agent.go#L1300,
so bindUnmountContainerRootfs should not be called if container's rootfs use devicemapper device
in https://github.com/kata-containers/runtime/blob/master/virtcontainers/container.go#L1123

Fixes: #937

Signed-off-by: Shukui Yang <keloyangsk@gmail.com>

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]
Thanks @keloyang.

File: src/runtime/virtcontainers/mount.go
Comment (amshinde) [inline src/runtime/virtcontainers/mount.go:358]: This check here should prevent bindUnmountContainerRootfs from being called. If it were being called, the bindUnmountContainerRootfs call would have returned an error which would have been propogated up the stack.
Are you seeing otherwise?

**Review by amshinde — DISMISSED** [review]

**Comment by jodh-intel** [comment]
Travis is failing with a strange error:

```
make: Entering directory '/home/travis/build/kata-containers/kata-containers/src/runtime'
go test -v -mod=vendor ./...
# github.com/kata-containers/kata-containers/src/runtime/virtcontainers
package github.com/kata-containers/kata-containers/src/runtime/virtcontainers (test)
	imports github.com/kata-containers/runtime/virtcontainers/types: cannot find package "." in:
	/home/travis/build/kata-containers/kata-containers/src/runtime/vendor/github.com/kata-containers/runtime/virtcontainers/types
FAIL	github.com/kata-containers/kata-containers/src/runtime/virtcontainers [setup failed]
```

**Comment by keloyang** [comment]
Ooops, my bad. thank you @jodh-intel 
```
-       "github.com/kata-containers/runtime/virtcontainers/types"
+       "github.com/kata-containers/kata-containers/src/runtime/virtcontainers/types"
```

**Comment by fidencio** [comment]
I'm really sorry for the disruption but in order to do the branch name changes announced [here](http://lists.katacontainers.io/pipermail/kata-dev/2021-January/001695.html) I'm adding the "do-not-merge" and the "needs-branch-adjustment" labels to the PR.

I'll add a new comment once the branch changes are done and once it happens I'd like to ask you to resubmit the PR against the new `main` branch, possibly linking to this PR in case you find the conversations / reviews already made important to proceed.

**Comment by fidencio** [comment]
Kata Containers repos are back to business.  Thanks a lot for your patience and I'm very happy to announce that there's no changes needed from your side as renaming the branches automagically reflected here.

If you notice something weird, some missing CI, please, do not hesitate on contacting us here or on #kata-dev on bit.ly/katacontainersslack.

Also, this is a good opportunity to have your PR rebased atop of `main`, to ensure your PR follow all the guidelines (https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format) enforced by this community, or even to re-evaluate whether the PR is currently needed.

Last but not least,  as the default branch has been renamed to `main`, if you have a local clone you'd need to update it and you do so by running:
```
git branch -m 2.0-dev main
git fetch origin
git branch -u origin/main main
```

Happy hacking!

File: src/runtime/virtcontainers/mount.go
Comment (jodh-intel) [reply src/runtime/virtcontainers/mount.go:358]: Any comment on this @keloyang?

**Comment by fidencio** [comment]
@keloyang, may I ask you to rebase your patches atop of current `main`?

**Comment by ariel-adam** [comment]
@keloyang any updates on this PR?
Would it be OK to close for now in case you will not have time to work on it?

**Comment by keloyang** [comment]
@fidencio @ariel-adam sorry for late update.

File: src/runtime/virtcontainers/mount.go
Comment (c3d) [reply src/runtime/virtcontainers/mount.go:389]: @keloyang Can you check with @amshinde about this scenario. I think what would be nice is we have an example of failure in the commit message.

File: src/runtime/virtcontainers/mount.go
Comment (c3d) [inline src/runtime/virtcontainers/mount.go:356]: Could you add a comment explaining what the test does. It is explicit from your commit message, but hard to deduce from the code. Or alternatively, add a log / debug message explaining why you exit there.

**Review by c3d — COMMENTED** [review]
Could you please add an example of failure to address @amshinde's comment?

**Comment by jodh-intel** [comment]
Retriggering the required "jenkins-ci-ubuntu-20-04" CI which appears to be "stuck"

/retest-ubuntu

**Comment by jodh-intel** [comment]
Sigh. 20.04 failed but the logs have been expired. Let's respin it...

/test-ubuntu.

**Comment by cmaf** [comment]
Hi @keloyang, if this PR is still relevant, could you please address comments from @amshinde and @c3d? It also needs a rebase.

**Comment by cmaf** [comment]
@keloyang any updates for this PR?

**Comment by cmaf** [comment]
ping @keloyang

**Comment by jodh-intel** [comment]
Marking as stale as this PR may have been abandoned.

**Comment by jodh-intel** [comment]
@keloyang - Can you rebase this so we can get it landed?

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by fidencio** [comment]
As this PR has been waiting for an update since May 2021, I'm closing it for now.

@keloyang, please, if you find some free cycles to work on this PR, please, just re-open it and let us know.
Thanks for the contribution and looking forward to seeing an up-to-date version of this PR.

**Pull request closed without merging**
